invoicer_project/static/js/login_request.js

```diff
@@ -0,0 +1,105 @@
+const host = "http://127.0.0.1:8000";
+const emailForm = document.getElementById("email_input_log_in_page");
+const passwordForm = document.getElementById("password_input_log_in_page");
+
+
+function validatePassword() {
+    return passwordForm.value !== '';
+}
+
+function validateEmail() {
+    let user_email = emailForm.value;
+    return !(user_email.includes(' ') || !(/^[a-zA-Z0-9.]{3,20}@(?:[a-zA-Z0-9]{2,20}\.){1,30}[a-zA-Z]{2,10}$/.test(user_email))) 
```

Comment:
I think we can change !(a || !b ) to (!a && b), don't we?

Reply:
No need in that, since Js && and || operators and both short circuit operators and will perform exact same amount of comparisons